Can't compile via tsc

There are some errors : 
 
Collision/b2BroadPhase.ts(111,16): error TS1251: Function declarations are not allowed inside blocks in strict mode when targeting 'ES3' or 'ES5'. Class definitions are automatically in strict mode.

Collision/b2DynamicTree.ts(655,9): error TS6133: 'freeCount' is declared but its value is never read.

Dynamics/b2Body.ts(308,9): error TS6133: 'found' is declared but its value is never read.

Particle/b2ParticleSystem.ts(4527,9): error TS6133: 'discarded' is declared but its value is never read.

--- a/Box2D/Box2D/Collision/b2BroadPhase.ts
+++ b/Box2D/Box2D/Collision/b2BroadPhase.ts
@@ -108,7 +108,7 @@ export class b2BroadPhase {
 
       // This is called from box2d.b2DynamicTree::Query when we are gathering pairs.
       // boolean b2BroadPhase::QueryCallback(int32 proxyId);
-      function QueryCallback(proxy: b2TreeNode): boolean {
+      const QueryCallback = function (proxy: b2TreeNode): boolean {
         // A proxy cannot form a pair with itself.
         if (proxy.m_id === queryProxy.m_id) {
           return true;

--- a/Box2D/Box2D/Collision/b2DynamicTree.ts
+++ b/Box2D/Box2D/Collision/b2DynamicTree.ts
@@ -652,11 +652,11 @@ export class b2DynamicTree {
     this.ValidateStructure(this.m_root);
     this.ValidateMetrics(this.m_root);
 
-    let freeCount: number = 0;
+    // let freeCount: number = 0;
     let freeIndex: b2TreeNode = this.m_freeList;
     while (freeIndex !== null) {
       freeIndex = freeIndex.parent; // freeIndex = freeIndex.next;
-      ++freeCount;
+      // ++freeCount;
     }
 
     ///b2Assert(this.GetHeight() === this.ComputeHeight());

--- a/Box2D/Box2D/Dynamics/b2Body.ts
+++ b/Box2D/Box2D/Dynamics/b2Body.ts
@@ -305,14 +305,14 @@ export class b2Body {
     ///b2Assert(this.m_fixtureCount > 0);
     let node: b2Fixture = this.m_fixtureList;
     let ppF: b2Fixture = null;
-    let found: boolean = false;
+    // let found: boolean = false;
     while (node !== null) {
       if (node === fixture) {
         if (ppF)
           ppF.m_next = fixture.m_next;
         else
           this.m_fixtureList = fixture.m_next;
-        found = true;
+        // found = true;
         break;
       }
 

--- a/Box2D/Box2D/Particle/b2ParticleSystem.ts
+++ b/Box2D/Box2D/Particle/b2ParticleSystem.ts
@@ -4524,7 +4524,7 @@ export class b2ParticleSystem {
     // Number of contacts processed for the current particle.
     let currentContacts = 0;
     // Output the number of discarded contacts.
-    let discarded = 0;
+    // let discarded = 0;
     let b2ParticleBodyContactRemovePredicate = function(contact: b2ParticleBodyContact): boolean {
       // This implements the selection criteria described in
       // RemoveSpuriousBodyContacts().
@@ -4541,7 +4541,7 @@ export class b2ParticleSystem {
       }
 
       if (currentContacts++ > k_maxContactsPerPoint) {
-        ++discarded;
+        // ++discarded;
         return true;
       }
 
@@ -4567,7 +4567,7 @@ export class b2ParticleSystem {
             return false;
           }
         }
-        ++discarded;
+        // ++discarded;
         return true;
       }
 

